File: python/1721_SwappingNodesInALinkedList.py
class Solution:
    def swapNodes(self, head: ListNode, k: int) -> ListNode:
        curr = front = temp = head
        i = 1
        while curr:
            if i == k:
                temp = curr
            elif i > k:
                front = front.next
            curr = curr.next
            i += 1
        
        temp.val, front.val = front.val, temp.val
        return head

File: swift/1721_SwappingNodesInALinkedList.swift
class Solution {
    func swapNodes(_ head: ListNode?, _ k: Int) -> ListNode? {
        var front = head, end = head, curr = head
        var i = 1
        while curr != nil {
            if i == k {
                front = curr
            } else if i > k {
                end = end?.next
            }
            i += 1
            curr = curr?.next
        }
        
        var tempVal = front!.val
        front!.val = end!.val
        end!.val = tempVal
        
        return head
    }
}
